plugin/better-whitespace.vim

```diff
@@ -7,58 +7,72 @@ if exists('g:loaded_better_whitespace_plugin')
-        execute 'let ' . a:var . ' = ' . string(a:value)
-    endif
+    let g:[a:var] = get(g:, a:var, a:value)
 endfunction
-
+"
 " Set the highlight color for trailing whitespaces
-call s:InitVariable('g:better_whitespace_ctermcolor', 'red')
-call s:InitVariable('g:better_whitespace_guicolor', '#FF0000')
+call s:InitVariable('better_whitespace_ctermcolor', 'red')
+call s:InitVariable('better_whitespace_guicolor', '#FF0000')
 
 " Operator for StripWhitespace (empty to disable)
-call s:InitVariable('g:better_whitespace_operator', '<leader>s')
+call s:InitVariable('better_whitespace_operator', '<leader>s')
 
 " Set this to enable/disable whitespace highlighting
-call s:InitVariable('g:better_whitespace_enabled', 1)
+call s:InitVariable('better_whitespace_enabled', 1)
 
 " Set this to match space characters that appear before or in-between tabs
-call s:InitVariable('g:show_spaces_that_precede_tabs', 0)
+call s:InitVariable('show_spaces_that_precede_tabs', 0)
 
 " Set this to disable highlighting on the current line in all modes
 " WARNING: This checks for current line on cursor move, which can significantly
 "          impact the performance of Vim (especially on large files)
-call s:InitVariable('g:current_line_whitespace_disabled_hard', 0)
+" WARNING: Ignored if g:current_line_whitespace_disabled_soft is set.
+call s:InitVariable('current_line_whitespace_disabled_hard', 0)
 
 " Set this to disable highlighting of the current line in all modes
 " This setting will not have the performance impact of the above, but
 " highlighting throughout the file may be overridden by other highlight
 " patterns with higher priority.
-call s:InitVariable('g:current_line_whitespace_disabled_soft', 0)
+call s:InitVariable('current_line_whitespace_disabled_soft', 0)
 
 " Set this to enable stripping whitespace on file save
-call s:InitVariable('g:strip_whitespace_on_save', 0)
+call s:InitVariable('strip_whitespace_on_save', 0)
 
 " Set this to enable stripping white lines at the end of the file when we
 " strip whitespace
-call s:InitVariable('g:strip_whitelines_at_eof', 0)
+call s:InitVariable('strip_whitelines_at_eof', 0)
+
+" Set this to enable user confirmation before stripping whitespace on file save
+call s:InitVariable('strip_whitespace_confirm', 1)
 
 " Set this to blacklist specific filetypes
-let default_blacklist=['diff', 'gitcommit', 'unite', 'qf', 'help', 'markdown']
-call s:InitVariable('g:better_whitespace_filetypes_blacklist', default_blacklist)
+call s:InitVariable('better_whitespace_filetypes_blacklist', ['diff', 'gitcommit', 'unite', 'qf', 'help', 'markdown'])
+
+" Skip empty (whitespace-only) lines for highlighting
+call s:InitVariable('better_whitespace_skip_empty_lines', 0)
+
+" Skip stripping whitespace on lines that have not been modified
+call s:InitVariable('strip_only_modified_lines', 0)
+
+" Skip stripping whitespace on files that have more lines than this variable
+call s:InitVariable('strip_disabled_on_large_files', 0)
```

Comment:
Thoughts on having a default value set for this? Maybe something like 1000 or 2000 lines?